packages/xod-project/src/gmath.js

```diff
@@ -6,14 +6,29 @@ export function findVertexesWithNoIncomingEdges(vertexes, edges) {
   return R.difference(vertexes, R.map(R.nth(1), edges));
 }
 
-export function hasIncomingEdges(vertex, edges) {
-  const edgeIncoming = R.compose(R.equals(vertex), R.nth(1));
-  return R.any(edgeIncoming, edges);
-}
+// Packs edges list in a data structure optimized for use in `sortGraph`
+//
+// :: [[Int, Int]] -> EdgesSet
```

Comment:
It will be nice to write somewhere that `EdgeSet` is `StrMap StrMap Boolean` :)

Reply:
@brusherru it's a implementation detail that I'd like to keep opaque.

@nkrkv I think access time won't improve by much, but `dissocEdge` could become a bit faster. Also, implementation will get a little bit more ugly😁 I'll try it.